File: subroutines/call-stack/calculate_pay.cs
/*

Note: This file is designed to be copied out and compiled on your machine.
In order for it to compile properly you need to ensure that the project name is the same as the "namespace" in this file.

To run this file you need to:
1. Copy the contents
2. Paste them into the C# IDE of your choice (Visual Studio, for example)
3. Compile the program
4. Run the program
*/

using System;

namespace Subroutines
{
    class CallStack
    {
        // Simple program to demonstrate the use of call stacks

        static void Main(string[] args) {            
            Console.WriteLine("Enter hours worked ");
            string hoursInput = Console.ReadLine();
            int hoursFloat = Convert.ToInt32(num1_input);
            Console.WriteLine("Enter hourly rate ");

            string rateInput = Console.ReadLine();
            int rateFloat = Convert.ToInt32(num2_input);
            int pay = CalculatePay(hoursFloat, rateFloat);
            Console.WriteLine(pay);            
        }

        // A simple pay calculation program
        static float CalculatePay(float h, float r) {
            float pay = h * r;
            return pay;
        }
    }
}

File: subroutines/call-stack/python/calculate_pay.py
# Raspberry Pi Foundation
# Developed to be used alongside Isaac Computer Science,
# part of the National Centre for Computing Education
# Usage licensed under CC BY-SA 4


def calculate_pay(h, r):
    """Performs a basic pay calculation"""
    pay =  h * r
    return pay


def main():
    """A simple pay calculator"""
    hours_input = input("Enter hours worked ")
    rate_input = input("Enter hourly rate ")
    hours = float(hours_input)
    rate = float(rate_input)
    pay = calculate_pay(hours, rate)
    print(f"Your pay is £{pay:.2f}")


if __name__ == '__main__':
    main()
